Taarafo.Core.Tests.Unit/Services/Foundations/GroupMemberships/GroupMembershipServiceTests.Exceptions.Add.cs

```diff
@@ -0,0 +1,211 @@
+        {
+            //given
+            GroupMembership someGroupMembership = CreateRandomGroupMembership();
+
+            var databaseUpdateException = new DbUpdateException();
+
+            var failedGroupMembershipStorageException =
+                new FailedGroupMembershipStorageException(databaseUpdateException);
+
+            var expectedGroupMembershipDependencyException =
+                new GroupMembershipDependencyException(failedGroupMembershipStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Throws(databaseUpdateException);
+
+            //when
+            ValueTask<GroupMembership> addGroupMembershipTask =
+                this.groupMembershipService.AddGroupMembershipAsync(someGroupMembership);
+
+            GroupMembershipDependencyException actualGroupMembershipDependencyException =
+                await Assert.ThrowsAsync<GroupMembershipDependencyException>(
+                    addGroupMembershipTask.AsTask);
+
+            //then
+            actualGroupMembershipDependencyException.Should().BeEquivalentTo(
+                expectedGroupMembershipDependencyException);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffset(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedGroupMembershipDependencyException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertGroupMembershipAsync(It.IsAny<GroupMembership>()),
+                    Times.Never);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowServiceExceptionOnAddIfDatabaseUpdateErrorOccursAndLogItAsync()
+        {
+            // given
+            GroupMembership someGroupMembership = CreateRandomGroupMembership();
+            var serviceException = new Exception();
+
+            var failedGroupMembershipServiceException =
+                new FailedGroupMembershipServiceException(serviceException);
+
+            var expectedGroupMembershipServiceException =
+                new GroupMembershipServiceException(failedGroupMembershipServiceException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Throws(serviceException);
+
+            // when
+            ValueTask<GroupMembership> addGroupMembershipTask =
+                this.groupMembershipService.AddGroupMembershipAsync(someGroupMembership);
+
+            // then
+            await Assert.ThrowsAsync<GroupMembershipServiceException>(() =>
+                addGroupMembershipTask.AsTask());
```

Comment:
you shuold compare expectedGroupMembershipServiceException  with actualGroupMembershipServiceException